If conStr is not defined, pass opts as a config object

tzaeru opened the pull request:
Heya,

I noticed that koa-pg doesn't allow config objects as used by node-postgres to be passed along. I added a simple check for passing 'opts' object to co-pg instead of opts.conStr if the latter is not defined. This seems to fix the issue, allowing full config objects to be passed along to co-pg instead of just connection strings.

tzaeru committed:
If conStr (connection string) is not defined, pass opts as a config object to co-pg. Goal to allow config objects as used and accepted by node-postgres.

chilts commented:
Might be better to be explicit about what it passes on instead of just passing itself on (in case any config options overlap).

Perhaps passing ` || opts.pg` instead of just ` || opts`? Does that sound better? In fact, we should just take out `opts.conStr` and change it to `opts.pg` and then you can pass either a connection string or a config object `{ ... }` which just gets passed straight through to `pg`.

I'm happy to merge whatever you settle on and perhaps something in the README to explain what is going on.

tzaeru committed:
Removed conStr, using pg member instead

tzaeru committed:
Update README.md

Typos

tzaeru committed:
Legacy support for conStr

tzaeru commented:
I renamed `conSt`r to `pg` where `pg` can be either a string or an object. The 'olden way of passing a simple connection string still works as well.

Also updated README to include a snippet on how the config object might be passed.

And added legacy support for anyone who happened to use `conStr` variable to pass a config object to co-pg. That's what I did at first, but it's a bit.. dirty!

chilts commented:
Awesome, thanks @tzaeru. :)

chilts commented:
Released as v1.0.0 to npm. Thanks again.
